test/e2e/framework/framework.go

```diff
@@ -60,3 +66,64 @@ func queryAndUpdateGlobalnetStatus() {
 		return true, "", nil
 	})
 }
+
+func (f *Framework) AwaitForGatewayWithStatus(cluster framework.ClusterIndex,
+	name string, status submarinerv1.HAStatus) *submarinerv1.Gateway {
+	gwClient := SubmarinerClients[cluster].SubmarinerV1().Gateways(framework.TestContext.SubmarinerNamespace)
+	gw := framework.AwaitUntil("await gateway ready",
+		func() (interface{}, error) {
+			resGw, err := gwClient.Get(name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return nil, nil
+			}
+			return resGw, err
+		},
+		func(result interface{}) (bool, string, error) {
+			if result == nil {
+				return false, "gateway not found yet", nil
+			}
+			gw := result.(*submarinerv1.Gateway)
+			if gw.Status.HAStatus != status {
+				return false, "", fmt.Errorf("Gateway %q showed up but has wrong status %q, expected %q",
+					gw.Name, gw.Status.HAStatus, status)
+			}
+			return true, "", nil
+		})
+	return gw.(*submarinerv1.Gateway)
+}
+
+func (f *Framework) AwaitForGatewayGone(cluster framework.ClusterIndex, name string) {
+	gwClient := SubmarinerClients[cluster].SubmarinerV1().Gateways(framework.TestContext.SubmarinerNamespace)
+	framework.AwaitUntil("await gateway gone",
+		func() (interface{}, error) {
+			if _, err := gwClient.Get(name, metav1.GetOptions{}); apierrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, nil
+		},
+		func(result interface{}) (bool, string, error) {
+			gone := result.(bool)
+			return gone, "", nil
```

Comment:
```suggestion
			return result.(bool), "", nil
```

Reply:
I prefer to keep it, It's added as self-documentation otherwise the lambda function was very obscure and I thought it was harder to understand the real intention of that bool